test/commands/trigger_test.cpp

```diff
@@ -0,0 +1,87 @@
+#include "common/harness.h"
+#include "commands/trigger.h"
+#include "parser/pg_trigger.h"
+#include "parser/postgresparser.h"
+#include "planner/create_plan.h"
+
+namespace peloton {
+namespace test {
+
+class TriggerTests : public PelotonTest {};
+
+TEST_F(TriggerTests, BasicTest) {
+  // Create statement
+  auto parser = parser::PostgresParser::GetInstance();
+
+  std::string query1 =
+      "CREATE TRIGGER check_update "
+      "BEFORE UPDATE OF balance ON accounts "
+      "FOR EACH ROW "
+      "WHEN (OLD.balance <> NEW.balance) "
+      "EXECUTE PROCEDURE check_account_update();";
+  auto stmt_list1 = parser.BuildParseTree(query1).release();
+  EXPECT_TRUE(stmt_list1->is_valid);
+  EXPECT_EQ(StatementType::CREATE, stmt_list1->GetStatement(0)->GetType());
+  auto create_trigger_stmt1 =
+      static_cast<parser::CreateStatement *>(stmt_list1->GetStatement(0));
+
+  // Create plans
+  const planner::CreatePlan plan1(create_trigger_stmt1);
+
+  commands::Trigger trigger1(plan1);
+  EXPECT_EQ("check_update", trigger1.GetTriggerName());
+  int16_t trigger_type1 = trigger1.GetTriggerType();
+  EXPECT_TRUE(TRIGGER_FOR_ROW(trigger_type1));
+  EXPECT_TRUE(TRIGGER_FOR_BEFORE(trigger_type1));
+  EXPECT_TRUE(TRIGGER_FOR_UPDATE(trigger_type1));
```

Comment:
Because the trigger we created in the test would fire "before any updates for each row in the target table"